src/lib/support/tests/TestStringCircularBuffer.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <string.h>
+
+#include <support/StringCircularBuffer.h>
+#include <support/UnitTestRegistration.h>
+
+#include <nlunit-test.h>
+
+namespace {
+
+using namespace chip;
+
+void TestPushPop(nlTestSuite * inSuite, void * inContext)
+{
+    uint8_t storage[10];
+    const unsigned char s[] = { 'T', 'e', 's', 't' };
```

Comment:
uint8_t, not unsigned char, please.  The other code is assuming uint8_t, for sure....  Same throughout this file.